offchain-modules/lib/src/transfer/to_eth.rs

```diff
@@ -29,14 +48,96 @@ pub fn burn(private_key: String, rpc_url: String) -> Result<String> {
         "tx_hash": hex::encode(tx.hash().as_slice()),
         "cell_typescript": cell_script,
     });
-    println!("{}", serde_json::to_string_pretty(&print_res).unwrap());
+    debug!("{}", serde_json::to_string_pretty(&print_res).unwrap());
     Ok(hex::encode(tx.hash().as_slice()))
 }
 
-pub fn parse_ckb_proof() -> Result<()> {
+pub fn unlock() -> Result<()> {
     todo!()
 }
 
-pub fn unlock() -> Result<()> {
-    todo!()
+pub fn get_add_ckb_headers_abi(headers: Vec<u8>) -> Bytes {
+    //TODO : addHeader is mock function for test feature which set header data in eth contract
+    let function = Function {
+        name: "addHeader".to_owned(),
+        inputs: vec![Param {
+            name: "_data".to_owned(),
+            kind: ParamType::Bytes,
+        }],
+        outputs: vec![],
+        constant: false,
+    };
+    function.encode_input(&[Token::Bytes(headers)]).unwrap()
+}
+
+pub fn parse_ckb_proof(tx_hash_str: &str, rpc_url: String) -> Result<CkbTxProof, String> {
```

Comment:
This Result is different with others in this crate. Should keep them the same type.
You can replace it with `Result<CkbTxProof>` and then replace all the `return Err(format!("Duplicated tx_hash {:#x}", tx_hash));` with `anyhow::bail!("Duplicated tx_hash {:#x}", tx_hash)`.

Reply:
> This Result is different with others in this crate. Should keep them the same type.
> You can replace it with `Result<CkbTxProof>` and then replace all the `return Err(format!("Duplicated tx_hash {:#x}", tx_hash));` with `anyhow::bail!("Duplicated tx_hash {:#x}", tx_hash)`.

done